test/native/libs/callsmalloc.c

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+
+__attribute__((visibility("default"))) void* call_malloc(size_t size) {
```

Comment:
Maybe you could reuse `Java_test_nativemem_Native_malloc` from `jnimalloc.c` instead of creating a new file?